### PR TITLE
Add jupyter notebooks to ecosystem checks

### PR DESCRIPTION
**Pull request by zanieb**
Implements https://github.com/astral-sh/ruff/pull/8873 via https://github.com/astral-sh/ruff/pull/9286

### CODE REVIEW

**Comment by zanieb** [comment]
Are these real issues with our formatter? cc @dhruvmanila

**Comment by konstin** [comment]
I checked some of the notebooks and they did contain syntax errors.

The error messages should have a cell number though.

**Comment by zanieb** [comment]
Yes I can confirm the huggingface notebooks are all over the place and have invalid syntax. We should improve the formatter error messages in those cases, but that's a separate concern. I've fixed the errors in https://github.com/zanieb/notebooks/commit/68cd6fa1a2831c5189f85257c13d691cb76292db but contributing upstream seems like a pain since the notebooks are synced from other repositories and the content is full of problems.

I opened a fix for the openai-cookbook error at https://github.com/openai/openai-cookbook/pull/964

**Comment by zanieb** [comment]
Hm very interesting, do we mutate IDs?

```diff
  "cells": [
   {
    "cell_type": "markdown",
-   "id": "3f738710-b5ef-46c4-b874-f13221ee9f76",
+   "id": "754e1af3-4775-4211-a476-d0f698c81cd4",
    "metadata": {
     "colab_type": "text",
     "id": "view-in-github"
```

**Comment by zanieb** [comment]
We create IDs if they do not exist at https://github.com/astral-sh/ruff/blob/c01bb0d485fb2bca14cf89aa4da2cdcd60fe0bbe/crates/ruff_notebook/src/notebook.rs#L155-L158

Presumably this just differs on each run of the Ruff formatter. Hm.

**Comment by zanieb** [comment]
Rebasing onto #9359

**Comment by zanieb** [comment]
See fix for ibis #9392

File: python/ruff-ecosystem/ruff_ecosystem/defaults.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
     Repository,
 )
 
+JUPYTER_NOTEBOOK_SELECT = "A,E703,F704,B015,B018,D100"
```
Comment (charliermarsh) [inline python/ruff-ecosystem/ruff_ecosystem/defaults.py:12]: Why these?

**Review by charliermarsh — APPROVED** [review]

File: python/ruff-ecosystem/ruff_ecosystem/defaults.py (hunk 1 above)
Comment (zanieb) [reply python/ruff-ecosystem/ruff_ecosystem/defaults.py:12]: @dhruvmanila chose them — they're rules that have different behavior for notebooks or something? Idk if we really need to select a subset though?

Comment (charliermarsh) [reply python/ruff-ecosystem/ruff_ecosystem/defaults.py:12]: Yeah I'd probably just expand them, but nbd either way.

Comment (zanieb) [reply python/ruff-ecosystem/ruff_ecosystem/defaults.py:12]: Like extend select? I'll just wait for Dhruv to be available — not crtical.

Comment (dhruvmanila) [reply python/ruff-ecosystem/ruff_ecosystem/defaults.py:12]: > they're rules that have different behavior for notebooks or something?

Yeah, I chose them for this exact reason but it's reasonable to just expand them and use `extend-select` if possible.

**Pull request merged**